### DISCUSSION TITLE
Why HTTPFlow.kill not always prevents connection to server?

### DISCUSSION THREAD

**Discussion by iocmet** [mitmproxy/mitmproxy#7315] [Q&A]
Test with chaining
![image](https://github.com/user-attachments/assets/2bb98b16-1771-4865-8023-1574cfdb1b03)

**Comment by Prinzhorn** [comment]
https://github.com/mitmproxy/mitmproxy/issues/4711

**Comment by iocmet** [comment]
Is this issue related to my problem? Not sure what transit means here, im just trying kill flow on HttpRequest event

**Reply by Prinzhorn** [reply]
Sorry I was just skimming over the code (pls don't post Screenshots of code in the future). In your case you likely just need `--set connection_strategy=lazy`

**Reply by iocmet** [reply]
Forgot about this problem because was waiting for response to question :/, thanks

**Comment by iocmet** [comment]
So it means not finished request?
